hbase-common/src/test/java/org/apache/hadoop/hbase/util/TestFutureUtils.java

```diff
@@ -80,6 +80,8 @@ public void testRecordStackTrace() throws IOException {
         startsWith("org.apache.hadoop.hbase.util.TestFutureUtils.testRecordStackTrace"));
       assertTrue(Stream.of(elements)
         .anyMatch(element -> element.toString().contains("--------Future.get--------")));
+    } catch (Throwable t) {
+      throw new AssertionError("Caught unexpected Throwable", t);
```

Comment:
What's this...

Reply:
This is to ensure that the test fails when the exception we expect isn't thrown. It would be simply `fail()` except that api doesn't let me retain the context provided by `t`.